### PR TITLE
[DOCS] Updates links to Elasticsearch documentation for 8.16.0

### PR DESCRIPTION
**Pull request by kosabogi**
:warning: Note: The documentation links containing `current` in the URL will not be live until version 8.16.0 is released.

:warning: DO NOT MERGE UNTIL 8.16.0 is released

### Overview:
This PR updates connector-related links and taxonomy to align with the changes discussed in the following issue: https://github.com/elastic/search-docs-team/issues/140#issuecomment-2401949619

### Changes:
- Updated connector-related links to reflect the latest documentation structure.
- Refined the taxonomy.
- Removed the document-level security folder, as its content has been incorporated into the relevant reference documents.
- Deleted all files under docs/reference (except README.md), as this information is now maintained in the official Elastic documentation.

### CODE REVIEW

**Comment by kosabogi** [comment]
> @kosabogi your PR description says this is a work in progress. If that's the case, can you please change this to a draft PR until it's ready to be reviewed?

Yes, thanks for the heads-up @navarone-feekery!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ items that may help during the review.-->
 - [ ] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
 - [ ] Considered corresponding documentation changes
 - [ ] Contributed any configuration settings changes to the configuration reference
-- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)
+- [ ] if you added or changed Rich Configurable Fields for an Elastic Managed Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/_connectors.ts)
```
Comment (leemthompo) [inline .github/PULL_REQUEST_TEMPLATE.md:27]: You can revert this one, because that code file name hasn't changed

**Review by leemthompo — COMMENTED** [review]
Looks good! Couple tiny comments, and we can do general cleanup by removing the reference and DLS docs that have been incorporated into the official docs already.

File: connectors/sync_job_runner.py [hunk 2]
```diff
@@ -231,7 +231,7 @@ async def _update_native_connector_authentication(self):
         """
         The connector secrets API endpoint can only be accessed by the Enterprise Search system role,
         so we need to use a client initialised with the config's username and password to first fetch
-        the API key for native connectors.
+        the API key for Elatic managed connectors.
```
Comment (leemthompo) [inline connectors/sync_job_runner.py:234]:
```suggestion
        the API key for Elastic managed connectors.
```

**Review by leemthompo — APPROVED** [review]
LGTM!

**Comment by jedrazb** [comment]
buildkite test this

**Comment by seanstory** [comment]
@kosabogi you're good to merge, if you can. If you can't, let us know, and we'll click the button. :)

**Comment by leemthompo** [comment]
I think we need to wait until 8.16.0 is live for this one :)

File: docs/REFERENCE.md [hunk 3]
```diff
@@ -1,51 +1,51 @@
...
+| [Salesforce](https://www.elastic.co/guide/en/elasticsearch/reference/current/es-connectors-salesforce.html) | **GA** | 8.12+ | 8.12+ | 8.11+ | 8.13+ | 8.13+ | [View code](https://github.com/elastic/connectors/tree/main/connectors/sources/salesforce.py) |
+| [ServiceNow](https://www.elastic.co/guide/en/elasticsearch/reference/current/es-connectors-servicenow.html) | **GA** | 8.10+ | 8.10+ | 8.11+ | 8.13+ | 8.13+ | [View code](https://github.com/elastic/connectors/tree/main/connectors/sources/servicenow.py) |
+| [Sharepoint Online](https://www.elastic.co/guide/en/elasticsearch/reference/current/es-connectors-sharepoint-online.html) | **GA** | 8.9+ | 8.9+ | 8.9+ | 8.9+ | 8.9+ | [View code](https://github.com/elastic/connectors/tree/main/connectors/sources/sharepoint_online.py) |
+| [Sharepoint Server](https://www.elastic.co/guide/en/elasticsearch/reference/current/es-connectors-sharepoint.html) | **Beta** | 8.15+ | - | 8.11+ | 8.13+ | 8.14+ | [View code](https://github.com/elastic/connectors/tree/main/connectors/sources/sharepoint_server.py) |
```
Comment (leemthompo) [inline docs/REFERENCE.md:48]:
```suggestion
| [Sharepoint Server](https://www.elastic.co/guide/en/elasticsearch/reference/current/es-connectors-sharepoint.html) | **Beta** | 8.15+ | - | 8.11+ | 8.13+ | 8.15+ | [View code](https://github.com/elastic/connectors/tree/main/connectors/sources/sharepoint_server.py) |
```

Comment (leemthompo) [reply docs/REFERENCE.md:48]: 🚗 related to https://github.com/elastic/elasticsearch/pull/116661/files#diff-36b646ff1a44f8213ed59dc5252846dedda7afb05d8e2fb5310754bd99077b5aR47

**Review by jedrazb — APPROVED** [review]

**Comment by jedrazb** [comment]
buildkite test this

**Pull request merged**